src/cow-react/modules/advancedOrders/hooks/useParseSlippage.ts

```diff
@@ -1,20 +1,23 @@
 import { Percent } from '@uniswap/sdk-core'
 import { useCallback } from 'react'
-import { MIN_SLIPPAGE_BPS, MAX_SLIPPAGE_BPS } from 'constants/index'
-import { SlippageError } from '@cow/modules/advancedOrders/state/advancedOrdersSettingsAtom'
+import { MIN_SLIPPAGE_BPS, MAX_SLIPPAGE_BPS, LOW_SLIPPAGE_BPS, HIGH_SLIPPAGE_BPS } from 'constants/index'
 import { useSetSlippage } from './useSetSlippage'
 
-export function useParseSlippage(
-  setSlippageInput: React.Dispatch<React.SetStateAction<string>>,
-  setSlippageError: React.Dispatch<React.SetStateAction<false | SlippageError>>
-) {
+type Props = {
+  setSlippageInput: React.Dispatch<React.SetStateAction<string>>
+  setSlippageError: React.Dispatch<React.SetStateAction<string | null>>
+  setSlippageWarning: React.Dispatch<React.SetStateAction<string | null>>
+}
+
+export function useParseSlippage({ setSlippageInput, setSlippageError, setSlippageWarning }: Props) {
   const setSlippage = useSetSlippage()
 
   return useCallback(
     (value: string) => {
       // populate what the user typed and clear the error
       setSlippageInput(value)
-      setSlippageError(false)
+      setSlippageError(null)
+      setSlippageWarning(null)
 
       if (value.length === 0) {
```

Comment:
I think the logic with parsing value must be inside `numberOfPartsValue, numberOfPartsError`TradeNumberInput`.

```
const onSlippageChange = (value: null | number) => {
    setSlippage(value === null ? 'auto' : value)
}

return < TradeNumberInput ...props, decimals={2} onChange={onSlippageChange}>
```

Reply:
Not sure if you meant inside the `AdvancedOrdersWidget` because slippage and number of parts use different parsing, so it can't be in `TradeNumberInput`. But this hook `useParseSlippage` has a lot of code and I don't want to put all that code inside the `AdvancedOrdersWidget` and the same for number of parts parsing. That is why I prefered to have 2 different components for slippage and number of parts